WebARonARKit/WebARonARKit/ViewController.m

```diff
@@ -564,25 +600,64 @@ - (NSString *)getPlanesString:(nonnull NSArray<ARAnchor *> *)anchors
                           -plane.extent.x / 2, 0.0, plane.extent.z / 2,
                           -plane.extent.x / 2, 0.0, -plane.extent.z / 2,
                           plane.extent.x / 2, 0.0, -plane.extent.z / 2];
-        if (i < anchors.count - 1) {
-            planeStr = [planeStr stringByAppendingString:@","];
-        }
+        planeStr = [planeStr stringByAppendingString:@","];
         result = [result stringByAppendingString:planeStr];
     }
-    result = [result stringByAppendingString:@"]"];
+    // Remove the last coma if there is any string
+    if (result != nil) {
+      result = [result substringToIndex:result.length - 1];
+      result = [result stringByAppendingString:@"]"];
+    }
     return result;
 }
 
-- (void) dispatchVRDisplayPlaneEvent:(NSString *)type planes:(NSString *)planes
+- (NSString *)getAnchorsString:(nonnull NSArray<ARAnchor *> *)anchors
+{
+  NSString *result = nil;
+  for (int i = 0; i < anchors.count; i++) {
+    if ([anchors[i] isKindOfClass:[ARPlaneAnchor class]] ||
+          [anchors[i] isKindOfClass:[ARFaceAnchor class]]) {
+      // We do not want Plane or Face anchors.
+      continue;
+    }
+    if (result == nil) {
+      result = @"[";
+    }
+    ARAnchor *anchor = (ARAnchor *)anchors[i];
+    matrix_float4x4 anchorTransform = anchor.transform;
+    const float *anchorMatrix = (const float *)(&anchorTransform);
+    NSString* jsAnchorId = objCAnchorIdsToJSAnchorIds[anchor.identifier.UUIDString];
+    NSString *anchorStr = [NSString stringWithFormat:
+                          @"{\"modelMatrix\":[%f,%f,%f,%f,%f,%f,%f,%f,"
+                          @"%f,%f,%f,%f,%f,%f,%f,%f],"
+                          @"\"identifier\":%@,",
+                          anchorMatrix[ 0], anchorMatrix[ 1], anchorMatrix[ 2],
+                          anchorMatrix[ 3], anchorMatrix[ 4], anchorMatrix[ 5],
+                          anchorMatrix[ 6], anchorMatrix[ 7], anchorMatrix[ 8],
+                          anchorMatrix[ 9], anchorMatrix[10], anchorMatrix[11],
+                          anchorMatrix[12], anchorMatrix[13], anchorMatrix[14],
+                          anchorMatrix[15],
+                          jsAnchorId];
+    anchorStr = [anchorStr stringByAppendingString:@","];
+    result = [result stringByAppendingString:anchorStr];
+  }
+  if (result != nil) {
+    result = [result substringToIndex:result.length - 1];
```

Comment:
Add comment about removing last comma (same as above)

Reply:
You are right.